### PR TITLE
Chapter13

### PR DESCRIPTION
**Pull request by NurAnisAqilah**
ready

### CODE REVIEW

**Comment by NurAnisAqilah** [comment]
ready

File: app/assets/stylesheets/microposts.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Microposts controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/microposts.scss]: remove file that never used

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
-  include SessionsHelper
+    include SessionsHelper
```
Comment (saobv-2223) [inline app/controllers/application_controller.rb]:
```suggestion
  include SessionsHelper
```

File: app/controllers/microposts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+class MicropostsController < ApplicationController
+    before_action :logged_in_user, only: %i(create destroy)
+    before_action :correct_user, only: :destroy
+  
+    def create;
```
Comment (saobv-2223) [inline app/controllers/microposts_controller.rb]:
```suggestion
    def create
```

File: app/controllers/microposts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    def micropost_params
+      params.require(:micropost).permit :content, :image
+    end
+  
+    def correct_user
+      @micropost = current_user.microposts.find_by id: params[:id]
+      redirect_to root_url unless @micropost
+    end
+  end
+
```
Comment (saobv-2223) [inline app/controllers/microposts_controller.rb]:
```suggestion
  end

```
Remove 2 spaces at the last line

File: app/controllers/static_pages_controller.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    @feed_items = current_user.feed.paginate(page: params[:page]).per `number`
+  end
+  
+  def help; end
+  
+  def about; end
+  
+  def contact; end
+end
+
```
Comment (saobv-2223) [inline app/controllers/static_pages_controller.rb]:
```suggestion
end

```
Same

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -20,7 +20,10 @@ def new
     @user = User.new
   end
 
-  def show; end
+  def show;
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
  def show
```

File: app/helpers/microposts_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module MicropostsHelper
+end
```
Comment (saobv-2223) [inline app/helpers/microposts_helper.rb]: Remove file that never used

File: app/models/user.rb [hunk 8]
```diff
@@ -1,75 +1,108 @@
...
-    update_attributes reset_digest: User.digest(reset_token), reset_sent_at: Time.zone.now
-  end
-
-  def send_password_reset_email
-    UserMailer.password_reset(self).deliver_now
+    self.activation_token = User.new_token
+    self.activation_digest = User.digest activation_token
   end
-end
+end
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
end

```
Add endline

**Comment by NurAnisAqilah** [comment]
ready

File: app/controllers/microposts_controller.rb [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  
+    def micropost_params
+      params.require(:micropost).permit :content, :image
+    end
+  
+    def correct_user
+      @micropost = current_user.microposts.find_by id: params[:id]
+      redirect_to root_url unless @micropost
+    end
+  end
```
Comment (saobv-2223) [inline app/controllers/microposts_controller.rb]: indenting 2 spaces, please

**Review by saobv-2223 — COMMENTED** [review]
fix comment and mege

**Pull request merged**